h2o-admissibleml/src/main/java/hex/Infogram/InfogramUtils.java

```diff
@@ -0,0 +1,239 @@
+
+    DKV.put(extractedFrame);
+    return extractedFrame;
+  }
+
+  public static String[] generateModelDescription(String[] topKPredictors, String[] sensitive_attributes) {
+    int numModel = topKPredictors.length+1;
+    String[] modelNames = new String[numModel];
+    int numPredInd = topKPredictors.length-1;
+    if (sensitive_attributes == null) { // contains only predictors
+      for (int index = 0; index < numPredInd; index++)
+        modelNames[index] = "Model built missing predictor "+topKPredictors[index];
+      modelNames[numPredInd] = "Full model built with all predictors";
+    } else {  // contains one predictor and all sensitive_attributes
+      for (int index = 0; index < numPredInd; index++)
+        modelNames[index] = "Model built with sensitive_features and predictor "+topKPredictors[index];
+      modelNames[numPredInd] = "Model built with sensitive_features only";
+    }
+    return modelNames;
+  }
+
+  /***
+   * Build model parameters for model specified in infogram_algorithm.  Any model specific parameters can be specified
+   * in infogram_algorithm_params.
+   * 
+   * @param trainingFrames
+   * @param infoParams
+   * @param numModels
+   * @param algoName
+   * @return
+   */
+  public static Model.Parameters[] buildModelParameters(Frame[] trainingFrames, Model.Parameters infoParams,
+                                                        int numModels, InfogramParameters.Algorithm algoName) {
+    ModelParametersSchemaV3 paramsSchema;
+    switch (algoName) {
+      case glm:
+        paramsSchema = new GLMV3.GLMParametersV3();
+        break;
+      case gbm:
+        paramsSchema = new GBMV3.GBMParametersV3();
+        break;
+      case drf:
+        paramsSchema = new DRFV3.DRFParametersV3();
+        break;
+      case deeplearning:
+        paramsSchema = new DeepLearningV3.DeepLearningParametersV3();
+        break;
+      default:
+        throw new UnsupportedOperationException("Unknown algo: " + algoName);
+    }
+    Model.Parameters[] modelParams = new Model.Parameters[numModels];
+    for (int index = 0; index < numModels; index++) {
+      modelParams[index] = (Model.Parameters) paramsSchema.fillFromImpl(infoParams).createAndFillImpl();
+      modelParams[index]._ignored_columns = null; // training frame contains only needed columns
+      modelParams[index]._train = trainingFrames[index]._key;
+    }
+    return modelParams;
+  }
+
+  public static ModelBuilder[] buildModelBuilders(Model.Parameters[] modelParams) {
+    int numModel = modelParams.length;
+    ModelBuilder[] modelBuilders = new ModelBuilder[numModel];
+    for (int index = 0; index < numModel; index++)
+      modelBuilders[index] = ModelBuilder.make(modelParams[index]);
+    return modelBuilders;
+  }
+  
+  public static void removeFromDKV(List<Key<Frame>> generatedFrameKeys) {
+    for (Key<Frame> oneFrameKey : generatedFrameKeys)
+        DKV.remove(oneFrameKey);
+  }
```

Comment:
looks like it is of public utilty: why not sharing this to `DKV` class?
```
static public Value[] remove(Key[] key) { … }
static public Value[] remove(Key[] key, Futures fs) { … }
```

Reply:
while I agree in general, I would prefer this kind of fine-tuning to be done later (in another revision)

I have 2 reasons: 
- this is an iteration 0 of the code, we need to first prove the concept works, is used by users, and invest the effort into cleaning up only after it is proven,
- if we modify h2o-core we have to make sure it is 100% correct and high-performance, simple forgotten call for blockForPending() can have catastrophic consequences where things appear to be working most of the time but sometimes break in unexpected ways

once we put something in core it needs to be correct and well tested, eg. not so recent addition to ArrayUtils - "indexOf" had an unexpected behavior (which was correct for the particular use-case) but didn't work in general - however - the method being in ArrayUtils was adopted by others and generated 3+ non-obvious bugs

I suggest keeping helper methods like this in "user-space" and focusing on making the algo work

Reply:
I will keep it as is for now.